src/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh

```diff
@@ -171,6 +171,11 @@ __vsc_update_cwd() {
 	builtin printf '\e]633;P;Cwd=%s\a' "$(__vsc_escape_value "$PWD")"
 }
 
+__vsc_update_prompt_height() {
+	__vsc_prompt_height="$(builtin printf "%s" "${PS1@P}" | wc -l)"
+	builtin printf '\e]633;P;PromptHeight=%s\a' "$(__vsc_escape_value "$__vsc_prompt_height")"
+}
```

Comment:
Neat idea, we probably have to make this a lower-bound for the prompt height though due to wrapping. See here how the VS Code thinks it's single line (bash/macOS):

<img width="546" alt="Screenshot 2024-04-16 at 4 10 31 PM" src="https://github.com/microsoft/vscode/assets/2193314/21ac4ea5-d6fa-4b8b-9e13-e84d9979d8b8">

Reply:
Could we set a condition that we only do this additional input adjustment if all the lines the prompt range (current + promptHeight) are not wrapped using xterm's `isWrapped` property?

Reply:
That will probably work 👍